packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -363,6 +408,55 @@ class LengthLimitingTextInputFormatter extends TextInputFormatter {
   /// composing is not allowed.
   final int? maxLength;
 
+  /// Determines how the [maxLength] limit should be enforced.
+  ///
+  /// Defaults to [MaxLengthEnforcement.enforced].
+  ///
+  /// {@macro flutter.services.textFormatter.maxLengthEnforcement}
+  final MaxLengthEnforcement? maxLengthEnforcement;
+
+  /// Return an effective [MaxLengthEnforcement] according the target platform.
+  ///
+  /// If [platform] was not specified, [defaultTargetPlatform] will be used.
+  ///
+  /// {@template flutter.services.textFormatter.effectiveMaxLengthEnforcement}
+  /// ### Platform specific behaviors
+  ///
+  /// Different platforms follow different behaviors by default, according to
+  /// their native behavior.
+  ///  * Android, Windows: [MaxLengthEnforcement.enforced]. The native behavior
```

Comment:
Can you help me to take a screenshot or a video? I've asked a friend who works on UWP and he saids the Microsoft PinYin IME will act lime this.

Reply:
This is the behavior in UWP.

![maxlength](https://user-images.githubusercontent.com/15884415/101555801-5709de00-39f4-11eb-8fc4-69a4ac88e3ab.gif)

Reply:
Thanks for the gif. I was struggling trying to record my screen. But from the gif it looks like the IME adds "placeholder" characters ("d" in the gif) to the text field while you're typing pinyin, and the candidates show up in a separate candidate view. Isn't that the same as iOS?

Reply:
You can try this with the path text field in windows explorer. The built-in IME behaves differently on that field, which held the composition by itself, like the Gboard in Android while typing CJKs. Basically it's enforced on Windows.

Reply:
The result that the GIF doesn't tell is 'd' this the last character before max length limit, and you cannot type any further.

Reply:
I think the special thing about Android Gboard is that it puts English inputs into a composition too which the windows IME doesn't seem to do. But since 

> The result that the GIF doesn't tell is 'd' this the last character before max length limit, and you cannot type any further.

setting the default to `enforced` makes sense to me too.